snapshot/git/gitdb/local_data.go

```diff
@@ -108,13 +109,76 @@ func (db *DB) ingestDir(dir string) (snapshot.ID, error) {
 	return v.ID(), nil
 }
 
+const tempRef = "refs/heads/scoot/__temp_for_writing"
+
+func (db *DB) ingestGitCommit(ingestRepo *repo.Repository, commitish string) (snapshot.ID, error) {
+	sha, err := ingestRepo.RunSha("rev-parse", "--verify", fmt.Sprintf("%s^{commit}", commitish))
+	if err != nil {
+		return "", fmt.Errorf("not a valid commit: %s, %v", commitish, err)
+	}
+
+	// Strategy:
+	// trying to move a commit from ingest to data
+	// delete the ref in the data.
+	// set the ref in the ingest.
+	// push from ingest to data.
+	// delete in both repos.
+	// TODO(dbentley): we could check if sha exists in our repo before ingesting
+
+	if _, err := db.dataRepo.Run("update-ref", "-d", tempRef); err != nil {
```

Comment:
I think if you try to push a ref that also exists in the other place, it could error... But I'm not sure. I guess this was just being careful...